### PR TITLE
created documentation files for best practices

### PR DESCRIPTION
**Pull request by gargajit**
Created this PR to document the Code best practices.

### CODE REVIEW

**Review by amitu — CHANGES_REQUESTED** [review]

**Review by Arpita-Jaiswal — CHANGES_REQUESTED** [review]

**Comment by amitu** [comment]
@gargajit comment when all the requested changes are done.

File: FASTN.ftd [hunk 1]
```diff
@@ -24,12 +24,14 @@ download-base-url: https://raw.githubusercontent.com/fifthtry/fastn.io/main
 -- fastn.auto-import: fastn.io/FASTN/my-ds as my-ds
 -- fastn.auto-import: fastn.io/assets as fastn-assets
 -- fastn.auto-import: fifthtry.github.io/common as common
+-- fastn.auto-import: fastn.io/utils as utils
```
[the remark this answers is not in the document]
Comment (gargajit) [reply FASTN.ftd:27]: At this point, yes, it was used only in one or two files, but in future, as the documentation of best-practices grows, I will keep on using the components inside this utils file.
I have already used it in 5 files in the latest commit.

File: FASTN.ftd [hunk 2]
```diff
@@ -92,8 +94,21 @@ download-base-url: https://raw.githubusercontent.com/fifthtry/fastn.io/main
...
+    document: /best-practices/formatting.ftd
+  - Import the packages: /import-at-top/
+    document: /best-practices/import-at-top.ftd
+  - Auto-import the packages: /auto-import/
+    document: /best-practices/auto-import.ftd
+  - Avoid CSS argument: /avoid-css-argument/
+    document: /best-practices/css-argument.ftd 
+  - Same argument and attribute types: /same-argument-attribute-type/
+    document: /best-practices/same-argument-attribute-type.ftd
+  -
```
[the remark this answers is not in the document]
Comment (gargajit) [reply FASTN.ftd:109]: this is outdated as I have fixed it in the latest commit.

File: FASTN.ftd (hunk 1 above)
[the remark this answers is not in the document]
Comment (amitu) [reply FASTN.ftd:27]: It is still not going to be almost all files. Right now there are 162 files in fastn.io. Almost all means what 90%? 3% can't be "almost all" right?

And whenever it start being used in more files add it to auto import. Do not violate the guidelines. Or update the guideline for whatever exception you think you are invoking.

File: best-practices/use-conditions.ftd [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+
+-- utils.code-display: `different-conditions-for-element-children`: Avoiding same conditions on element and it's children
+id: different-conditions-for-element-children
+
+It is not recommended to create same conditions on element and it's children.
+This approach adds an unnecessary line of code and can make the `ftd` code more
+difficult to read and maintain.
+
+Instead, the recommended approach is to include the condition only on the 
+element and then include any necessary child within that element.
```
Comment (Arpita-Jaiswal) [inline best-practices/use-conditions.ftd:93]:
```suggestion
Instead, it is recommended to apply conditions only to the element
itself, and omit applying conditions to its children.
```

File: utils.ftd [hunk 4]
```diff
@@ -190,3 +196,141 @@ width: fill-container
...
+width: fill-container
+id: $code-display.id
+
+-- ftd.column:
+width: fill-container
+padding-horizontal.px: 14
+padding-vertical.px: 12
+background.solid: $inherited.colors.background.code
+;; border-top-left-radius.px: 4
+;; border-top-right-radius.px: 4
```
Comment (Arpita-Jaiswal) [inline utils.ftd:224]:
```suggestion
```

File: utils.ftd [hunk 5]
```diff
@@ -190,3 +196,141 @@ width: fill-container
...
+
+-- ftd.row:
+spacing.fixed.px: 16
+align-content: center
+
+-- ftd.column:
+width.fixed.px: 8
+height.fixed.px: 8
+background.solid: $inherited.colors.custom.one
+;; border-radius.px: 12
```
Comment (Arpita-Jaiswal) [inline utils.ftd:234]:
```suggestion
```

Remove comments

**Review by Arpita-Jaiswal — CHANGES_REQUESTED** [review]

**Pull request merged**